src/TSTransformer/nodes/statements/transformReturnStatement.ts

```diff
@@ -14,30 +15,58 @@ function isTupleReturningCall(state: TransformState, tsExpression: ts.Expression
 	);
 }
 
-export function transformReturnStatementInner(state: TransformState, returnExp: ts.Expression) {
-	let expression: luau.Expression | luau.List<luau.Expression> = transformExpression(state, skipDownwards(returnExp));
-	if (isLuaTupleType(state)(state.getType(returnExp)) && !isTupleReturningCall(state, returnExp, expression)) {
-		if (luau.isArray(expression)) {
-			expression = expression.members;
-		} else {
-			expression = luau.call(luau.globals.unpack, [expression]);
+function isTupleMacro(state: TransformState, expression: ts.Expression) {
+	if (ts.isCallExpression(expression)) {
+		const expType = state.typeChecker.getNonOptionalType(state.getType(expression.expression));
+		const symbol = getFirstDefinedSymbol(state, expType);
+		if (symbol && symbol === state.services.macroManager.getSymbolOrThrow(SYMBOL_NAMES.$tuple)) {
+			return true;
+		}
+	}
+	return false;
+}
+
+export function transformReturnStatementInner(
+	state: TransformState,
+	returnExp: ts.Expression,
+): luau.List<luau.Statement> {
+	const result = luau.list.make<luau.Statement>();
+
+	let expression: luau.Expression | luau.List<luau.Expression>;
+
+	if (ts.isCallExpression(returnExp) && isTupleMacro(state, returnExp)) {
+		const [args, prereqs] = state.capture(() => ensureTransformOrder(state, returnExp.arguments));
+		luau.list.pushList(result, prereqs);
+		expression = luau.list.make(...args);
```

Comment:
I feel like this prereqs manual case is not necessary. Won't it already automatically go to the correct position if just allowed to stay as prereqs?

Reply:
Originally, I only designed the prereqs system for expressions and not statements. It's possible some places in the code base _do_ emit prereqs from statement transforms.. but I'm not 100% sure it works correctly.